usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -2844,6 +2844,13 @@ static CK_RV import_aes_xts_key(STDLL_TokData_t *tokdata, SESSION *sess,
         return rc;
     }
 
+    if (memcmp(attr->pValue,
+               ((CK_BYTE *)attr->pValue) + attr->ulValueLen / 2,
+               attr->ulValueLen / 2) == 0) {
```

Comment:
Is at this point already checked that attr->ulValueLen is an even number and also is an allowed even number?

Reply:
Yes, aes_validate_attribute() in key.c checks for that during attribute validation, which was run before.